packages/commonwealth/client/scripts/views/pages/view_thread/index.tsx

```diff
@@ -629,73 +632,75 @@ class ViewThreadPage
-              : []),
-            ...(this.polls?.length > 0 ||
-            (isAuthor && (!app.chain?.meta?.adminOnlyPolling || isAdmin))
-              ? [
-                  {
-                    label: 'Polls',
-                    item: (
-                      <div class="cards-column">
-                        {[
-                          ...new Map(
-                            this.polls?.map((poll) => [poll.id, poll])
-                          ).values(),
-                        ].map((poll: Poll) => {
-                          return <ThreadPollCard poll={poll} />;
-                        })}
-                        {isAuthor &&
-                          (!app.chain?.meta?.adminOnlyPolling || isAdmin) && (
-                            <ThreadPollEditorCard
+          sidebarComponents={
+            [
+              ...(showLinkedSnapshotOptions || showLinkedThreadOptions
+                ? [
+                    {
+                      label: 'Links',
+                      item: (
+                        <div class="cards-column">
+                          {showLinkedSnapshotOptions && (
+                            <LinkedProposalsCard
+                              onChangeHandler={(
+                                stage: ThreadStageType,
+                                chainEntities: ChainEntity[],
+                                snapshotProposal: SnapshotProposal[]
+                              ) => {
+                                thread.stage = stage;
+                                thread.chainEntities = chainEntities;
+                                if (app.chain?.meta.snapshot.length) {
+                                  thread.snapshotProposal =
+                                    snapshotProposal[0]?.id;
+                                }
+                                app.threads.fetchThreadsFromId([
+                                  thread.identifier,
+                                ]);
+                                m.redraw();
+                              }}
                               thread={thread}
-                              threadAlreadyHasPolling={!this.polls?.length}
+                              showAddProposalButton={isAuthor || isAdminOrMod}
+                            />
+                          )}
+                          {showLinkedThreadOptions && (
+                            <LinkedThreadsCard
+                              threadlId={thread.id}
+                              allowLinking={isAuthor || isAdminOrMod}
                             />
                           )}
-                      </div>
-                    ),
-                  },
-                ]
-              : []),
-          ]}
+                        </div>
+                      ),
+                    },
+                  ]
+                : []),
+              ...(this.polls?.length > 0 ||
+              (isAuthor && (!app.chain?.meta?.adminOnlyPolling || isAdmin))
```

Comment:
These changes safe? Seem ok just want to make sure.

Reply:
Yeah, IIRC that was just to cast the array to `SidebarComponents`.